pkg/http/api_test.go

```diff
@@ -2948,17 +2952,15 @@ func TestV1Alpha1DistributedLock(t *testing.T) {
 		b, _ := json.Marshal(&req)
 
 		resp := fakeServer.DoRequest("POST", apiPath, b, nil)
-		assert.Equal(t, 200, resp.StatusCode)
+		assert.Equal(t, 400, resp.StatusCode)
 
 		// assert
-		assert.NotNil(t, resp.JSONBody)
-		rspMap := resp.JSONBody.(map[string]interface{})
-		assert.NotNil(t, rspMap)
-		assert.Equal(t, float64(3), rspMap["status"])
+		assert.Contains(t, string(resp.RawBody), "ERR_MALFORMED_REQUEST")
+		assert.Contains(t, string(resp.RawBody), "ResourceId is empty in lock store store1")
 	})
 
 	t.Run("Unlock with invalid resource id that returns 500", func(t *testing.T) {
```

Comment:
shouldn't this also return `404`?

Reply:
This was in the original API design, and I didn't change it.

I agree with you that this is not the correct status code, but changing that would require reworking the entire APIs down to the components. Right now, they return an error in all cases where a lock can't be released.

I'll open an issue to track that.

Reply:
https://github.com/dapr/dapr/issues/3549#issuecomment-1487129276